README.md

```diff
@@ -47,23 +111,69 @@ setContextLogger(logger);
 
 describe('app code', () => {
 	it('returns 200', async () => {
-	    const { context, err, propertyBag, } = await stubContext(functionToTest);
+        const context = await runStubFunctionFromBindings(functionToTest, [
+            { type: 'httpTrigger', name: 'req', direction: 'in', data: createHttpTrigger('GET', 'http://example.com') },
+            { type: 'http', name: 'res', direction: 'out' },
+        ], new Date());
 	    expect(context).to.have.nested.property('res.status', 200);
 	});
 	it('returns 200 in promise/a+ style', (done) => {
-		stubContext(functionToTest)
-			.then(({ context, err, propertyBag }) => {
+		runStubFunctionFromBindings(functionToTest, [
+            { type: 'httpTrigger', name: 'req', direction: 'in', data: createHttpTrigger('GET', 'http://example.com') },
+            { type: 'http', name: 'res', direction: 'out' },
+        ], new Date())
+			.then((context) => {
 				expect(context).to.have.nested.property('res.status', 200);
 				done();
 			})
 			.catch(done);
 	});
-	it('change trigger values before calling your function under test', async () => {
-		const { context } = await stubContext((context, req, ...otherTriggers) => {
-		    req.body = { 'helpful': 'test object' };
-		    return functionToTest(context, req, ...otherTriggers);
-		});
-		expect(context).to.have.nested.property('res.body.helpful', 'test object');
-	});
+    it('supports $return values', async () => {
+        const response = await runStubFunctionFromBindings(functionToTest, [
+            { type: 'httpTrigger', name: 'req', direction: 'in', data: createHttpTrigger('GET', 'http://example.com') },
+            { type: 'http', name: '$return', direction: 'out' },
+        ], new Date());
+        expect(response).to.have.nested.property('status', 200);
+    });
+});
+```
+
+### Queue examples
+
+```js
+const { 
+    runStubFunctionFromBindings,
+    createQueueTrigger,
+    createQueueTriggerFromMessage,
+} = require('stub-azure-function-context');
+const functionToTest = require('./function-to-tes');
+const { promisify } = require('util');
+const { createQueueService, QueueMessageEncoder } = require('azure-storage');
+
+describe('queue triggered message', () => {
+    let queue;
+    before('set up queue', () => {
```

Comment:
shouldn't this be async if you're running promise stuff?

Reply:
It returns a promise, so the use of `async` is not needed.

Reply:
Don't you therefore need to pass done or is this a new feature I've missed?

Reply:
an `async` function is just one that returns a Promise. There is no internal distinction between:

```js
function myFunction() {
    return Promise.resolve();
}

async function myFunction() {
}
```

Mocha supports callbacks which are promises so no need for `done` nor for `async`. The `done` callback in mocha is really for when you're in callback hell / not using promises.